documentation/modules/managing/con-broker-config-properties.adoc

```diff
@@ -0,0 +1,520 @@
+
+By default, Kafka is enabled for automatic partition leader rebalancing based on a periodic check of leader distribution. That is, Kafka checks to see if the preferred leader is the current leader.
+A rebalance ensures that leaders are evenly distributed across brokers and brokers are not overloaded.
+You can control the frequency, in seconds, of the rebalance check and the maximum percentage of imbalance allowed for a broker before a rebalance is triggered.
+
+[source,env]
+----
+#...
+auto.leader.rebalance.enable=true
+leader.imbalance.check.interval.seconds=300
+leader.imbalance.per.broker.percentage=10
+#...
+----
+
+The percentage imbalance for a broker is the gap between the current number of partition leaders it holds and the number of partitions which are preferred leaders. You can set the percentage to zero to ensure that preferred leaders are always elected.
+
+If the checks for automated rebalances need more control, you can disable it. You can then choose to trigger a rebalance.
+Alternatively, you can xref:cruise-control-concepts-str[use Cruise Control for Strimzi] to change partition leadership and rebalance replicas across your Kafka cluster in a more intelligent way.
+
+NOTE: The Grafana dashboards provided with Strimzi show metrics for under-replicated partitions and partitions that do not have an active leader.
+
+[id='con-broker-config-properties-unclean-{context}']
+== Unclean leader election
+
+Leader election to an in-sync replica is considered clean because it guarantees no loss of data. But what if there is no in-sync replica to take on leadership?
+If a minimum number of in-sync replicas is not set, and there are no followers in sync with the partition leader when its hard drive fails irrevocably, data is already lost.
+Not only that, but a new leader cannot be elected because there are no in-sync followers.
+
+You can configure how Kafka handles leader failure:
+
+[source,env]
+----
+# ...
+unclean.leader.election.enable=false
+# ...
+----
+
+Unclean leader election is disabled by default, which means that out-of-sync replicas cannot become leaders.
+Kafka waits until the original leader is back online before messages are picked up again.
+Unclean leader election means out-of-sync replicas can become leaders, but you risk losing messages.
+The choice you make depends on whether your requirements favor availability or durability. If you cannot afford the risk of data loss, then leave the default configuration.
+
+== Avoiding unnecessary consumer group rebalances
+
+If new consumers are joining a consumer group that is making requests on a broker, you can add a delay so that unnecessary rebalances to the broker are avoided:
```

Comment:
I don't think we really need the "that is making requests on a broker" part. And this isn't really _adding_ a delay, rather the delay is an intrinsic part of how the coordinator works. 

I think something like this is all that needs to be said:

When a group is being created, or is rebalancing, this is the amount of time that the coordinator will wait for members to join the group. The longer this is, the more likely it is that all the members will join in time (thus avoiding a rebalance), but the longer the group will be prevented from actually consuming.

Reply:
Changed to:

_When creating or rebalancing a group, the delay is the amount of time that the coordinator waits for members to join. The longer the delay, the more likely it is that all the members will join in time and avoid a rebalance.
But the delay also prevents the group from consuming until the period has ended._